src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -402,21 +403,9 @@ public function display()
             $this->setRealFormat($this->internalFormat ?: $this->determineBestFormat());
         }
 
-        $this->overwrite(preg_replace_callback("{%([a-z\-_]+)(?:\:([^%]+))?%}i", function ($matches) {
-            if ($formatter = $this::getPlaceholderFormatterDefinition($matches[1])) {
-                $text = call_user_func($formatter, $this, $this->output);
-            } elseif (isset($this->messages[$matches[1]])) {
-                $text = $this->messages[$matches[1]];
-            } else {
-                return $matches[0];
-            }
-
-            if (isset($matches[2])) {
-                $text = sprintf('%'.$matches[2], $text);
-            }
-
-            return $text;
-        }, $this->format));
+        $line = $this->buildLine();
+        $line = $this->adjustLineWidthToTerminalWidth($line);
```

Comment:
This looks a bit strange: first I build a potentially wrong line (`buildLine()`) and then I fix the line if needed before displaying it (`adjustLineWidthToTerminalWidth()`). Shouldn't the `adjustLineWidthToTerminalWidth()` details be handled privately?